### ISSUE TITLE
`get` is not a function

### ISSUE THREAD

**Issue by panthony** [rmachielse/ember-enum#2]
Hi there,

When I try to create a new model with the following attribute:

```js
  currency: attr('enum', {
    options: [
      'EUR',
      'USD'
    ],
    defaultValue: 'EUR'
  }),
```

I have a `get` is not a function in the serialize method:

```js
  serialize(deserialized) {
    if (deserialized) {
      return deserialized.get('value');
    }
  }
```

The `deserialized` parameter is a string with `EUR` as value, not an Ember.Object.

I'm using ember-data 2.9, is it a breaking change? Or there are cases where deserialized may be an object ?

**Comment by rmachielse** [comment]
@panthony yeah unfortunately there is currently no way in ember to set up the enum object before it has been saved to the backend. If you do want to you should use something like this I think:

```javascript
import Model from 'ember-data/model';
import attr from 'ember-data/attr';
import Enum from 'ember-enum/enum';

export default Model.extend({
  currency: attr('enum', {
    options: [
      'EUR',
      'USD'
    ],
    defaultValue: Enum.create({
      value: 'EUR',
      options: [
        'EUR',
        'USD'
      ]
    })
  })
});
```

**Comment by rmachielse** [comment]
@panthony I'll investigate if I can find a way to make it work the way it's documented, so you don't have to repeat your options etc.

**Comment by panthony** [comment]
In the meantime I made the following workaround:

https://github.com/rmachielse/ember-enum/compare/master...cogniteev:fix/serialize-of-string

Since the value is already a string there is nothing to deserialize really so I just pass the value through.

**Comment by rmachielse** [comment]
@panthony I see, that will prevent the error, but you probably can't use the attribute as an enum object until it's been persisted in that case

**Comment by panthony** [comment]
@rmachielse You are probably right, I did not try to use it in a component yet.

Maybe a new `defaultOption` value which take one of the `options` value and create the default value for this attribute properly ?

I'll take a look on Monday.

**Comment by Gowiem** [comment]
Just ran into this. Would be cool to get a fix in as it seems like it'd be simple enough. I'll see if I can look into it at some point this week.

**Comment by Gowiem** [comment]
@rmachielse Looks like you fixed this in [`utils/enum.js`](https://github.com/Gowiem/ember-enum/blob/master/addon/utils/enum.js#L4), but that is not in the package on NPM. Mind doing a deploy of the package to get that out there?

**Comment by rmachielse** [comment]
@Gowiem sure, I have cut a release now. Although I am still uncertain about wether this package is using the right approach and should be continued.

**Comment by Gowiem** [comment]
@rmachielse Thanks for the quick turn around man! v0.2.0 is working for me with using that util method. 

This package seems to be working out of the box with the [Rail's `ActiveRecord::Enum` type](http://api.rubyonrails.org/classes/ActiveRecord/Enum.html), so I feel like it is useful at least in the context of Rails. Don't discontinue 😅

**Comment by rmachielse** [comment]
@Gowiem ok, good to know! The biggest problem is that it doesn't do dirt tracking on enum attributes, I haven't found a solution for that yet
